crates/sui-protocol-config/src/lib.rs

```diff
@@ -413,28 +428,32 @@ impl ProtocolConfig {
 
     /// Convenience to get the constants at the current minimum supported version.
     /// Mainly used by client code that may not yet be protocol-version aware.
-    pub fn get_for_min_version() -> &'static Self {
+    pub fn get_for_min_version() -> Arc<Self> {
         if Self::load_poison_get_for_min_version() {
             panic!("get_for_min_version called on validator");
         }
 
-        static CONSTANTS: Lazy<ProtocolConfig> =
-            Lazy::new(|| ProtocolConfig::get_for_version(ProtocolVersion::MIN));
+        thread_local! {
```

Comment:
this and next one are very confusing and I do not remember well how thread local work in Rust.
Do I understand it that every thread that calls in here will stash away a `ProtocolConfig` for caching so that when the same thread tried to create one it will be there?
what happens if the same thread calls `get_for_min_version` and then `get_for_max_version`?
is there an issue?